File: apps/web/src/types/workspaces.ts
export type WorkspaceType =
  | 'local_existing'
  | 'local_created'
  | 'git_clone_pending'
  | 'manual'

export type WorkspaceStatus =
  | 'registered'
  | 'ready'
  | 'missing'
  | 'invalid'
  | 'archived'

export interface WorkspaceCreate {
  code_repository_id?: string | null
  name: string
  root_path?: string | null
  workspace_type?: WorkspaceType
  description?: string | null
  create_directory?: boolean
}

export interface WorkspaceUpdate {
  name?: string | null
  description?: string | null
  status?: WorkspaceStatus | null
  code_repository_id?: string | null
}

export interface Workspace {
  id: string
  project_id: string
  code_repository_id: string | null
  name: string
  root_path: string
  workspace_type: WorkspaceType
  status: WorkspaceStatus
  description: string | null
  created_at: string
  updated_at: string
  last_inspected_at: string | null
  error_message: string | null
}

export interface WorkspaceInspection {
  workspace_id: string
  exists: boolean
  is_directory: boolean
  is_git_repo: boolean
  current_branch: string | null
  dirty: boolean
  file_count_estimate: number
  blocked_path_hits: string[]
  notes: string[]
}

File: services/api/app/models/workspaces.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

WorkspaceType = Literal["local_existing", "local_created", "git_clone_pending", "manual"]
WorkspaceStatus = Literal["registered", "ready", "missing", "invalid", "archived"]


class WorkspaceCreate(BaseModel):
    code_repository_id: str | None = None
    name: str
    root_path: str | None = None
    workspace_type: WorkspaceType = "local_created"
    description: str | None = None
    create_directory: bool = True


class WorkspaceUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    status: WorkspaceStatus | None = None
    code_repository_id: str | None = None


class Workspace(BaseModel):
    id: str
    project_id: str
    code_repository_id: str | None = None
    name: str
    root_path: str
    workspace_type: WorkspaceType
    status: WorkspaceStatus
    description: str | None = None
    created_at: datetime
    updated_at: datetime
    last_inspected_at: datetime | None = None
    error_message: str | None = None


class WorkspaceInspection(BaseModel):
    workspace_id: str
    exists: bool
    is_directory: bool
    is_git_repo: bool
    current_branch: str | None = None
    dirty: bool = False
    file_count_estimate: int = 0
    blocked_path_hits: list[str] = []
    notes: list[str] = []
